solidity/contracts/converter/BancorConverter.sol

```diff
@@ -912,21 +933,27 @@ contract BancorConverter is IBancorConverter, TokenHandler, SmartTokenController
         private
         returns (uint256)
     {
-        uint256[] memory reserveBalances = getBalances(_reserveTokens);
+        syncReserveBalances();
+        reserves[ETH_RESERVE_ADDRESS].balance = reserves[ETH_RESERVE_ADDRESS].balance.sub(msg.value);
+
         IBancorFormula formula = IBancorFormula(addressOf(BANCOR_FORMULA));
-        uint256 supplyAmount = getMinShare(_totalSupply, reserveBalances, _reserveAmounts);
+        uint256 supplyAmount = getMinShare(_totalSupply, _reserveTokens, _reserveAmounts);

```

Comment:
does getMinShare return the TOTAL supply should be issued?

Reply:
@yudilevi : It happens to do that as well. But the purpose of this function is to find the smallest share invested by the caller.

Reply:
Sure, just wanted to make sure that it doesn't return the supply per reserve and then we need to multiply it by the number of reserves in order to get the final supply.

Reply:
@yudilevi: No, it calculates the exact amount which will be issued to a user who adds liquidity to a non-empty pool.

Reply:
cool